src/Data/PQueue/Prio/Max.hs

```diff
@@ -298,9 +305,24 @@ foldlWithKey f z0 (MaxPQ q) = Q.foldlWithKey (\z -> f z . unDown) z0 q
 -- (@'traverseWithKey' f q == 'fromDescList' <$> 'traverse' ('uncurry' f) ('toDescList' q)@)
 --
 -- If you do not care about the /order/ of the traversal, consider using 'traverseWithKeyU'.
+--
+-- If you are working in a strict monad, consider using 'mapMWithKey'.
 traverseWithKey :: (Ord k, Applicative f) => (k -> a -> f b) -> MaxPQueue k a -> f (MaxPQueue k b)
 traverseWithKey f (MaxPQ q) = MaxPQ <$> Q.traverseWithKey (f . unDown) q
 
+-- | A version of 'traverseWithKey' designed to be efficient for strict monads
+-- such as 'IO' and strict @State@.
+mapMWithKey :: (Ord k, Monad m) => (k -> a -> m b) -> MaxPQueue k a -> m (MaxPQueue k b)
+mapMWithKey f = go empty
+  where
+    go acc q = acc `seq`
+      case maxViewWithKey q of
+        Nothing           -> pure acc
+        Just ((k, a), q') -> do
+          b <- f k a
+          let !acc' = insertMin' k b acc
```

Comment:
```suggestion
          let !acc' = insertMax' k b acc
```
`k` is the maximum key, so this should use `insertMax'`, no?

Reply:
It's the maximum of (what remains of) the original queue. It's the minimum of the accumulator.

Reply:
I see, you're right.